Src/java/cql-to-elm/src/main/java/org/cqframework/cql/cql2elm/CqlTranslator.java

```diff
@@ -328,12 +349,22 @@ private CqlTranslator(NamespaceInfo namespaceInfo, ANTLRInputStream is, ModelMan
 
     private CqlTranslator(NamespaceInfo namespaceInfo, ANTLRInputStream is, ModelManager modelManager,
                           LibraryManager libraryManager, UcumService ucumService, CqlTranslatorOptions options) {
+        this(namespaceInfo, null, is, modelManager, libraryManager, ucumService, options);
+    }
+
+    private CqlTranslator(NamespaceInfo namespaceInfo, VersionedIdentifier sourceInfo, ANTLRInputStream is, ModelManager modelManager,
+                          LibraryManager libraryManager, UcumService ucumService, CqlTranslatorOptions options) {
+        this.sourceInfo = sourceInfo;
         this.namespaceInfo = namespaceInfo;
         this.modelManager = modelManager;
         this.libraryManager = libraryManager;
         this.ucumService = ucumService;
         this.errorLevel = options.getErrorLevel();
 
+        if (this.sourceInfo == null) {
+            sourceInfo = new VersionedIdentifier().withId("Anonymous").withSystem("text/cql");
```

Comment:
If you were to translate a library from an input stream defined as:

library ExpressionLibrary
define Test: 2+2
define -syntax error!

You'd end up with a source info of "Anonymous" which is counter-intuitive. Currently, the code looks up the filename of the CQL source if one is available. It's not generally possible make sure there's always a library identifier present (as the SyntaxErrorWithNoLibrary shows). But would it be more user friendly to attempt to parse the library name from the InputStream for cases where that's possible?

Reply:
Good point, and I thought I had that covered, but it doesn't work like I expected, so I added some tests and some code to pull it from what's parsed so far if it's available.